feat(html-reporter): recognize video attachment by the contentType

RemigiuszDudek opened the pull request:
Closes https://github.com/microsoft/playwright/issues/33073.

RemigiuszDudek committed:
feat(custom-video-names): recognize video attachment by the contentType

dgozman commented on packages/html-reporter/src/testResultView.tsx:
```diff
@@ -70,7 +70,7 @@ export const TestResultView: React.FC<{
     const attachments = result?.attachments || [];
     const screenshots = new Set(attachments.filter(a => a.contentType.startsWith('image/')));
-    const videos = attachments.filter(a => a.name === 'video');
+    const videos = attachments.filter(a => a.contentType === 'video/webm');
```
We should allow all video types:

```suggestion
    const videos = attachments.filter(a => a.contentType.startsWith('video/'));
```

RemigiuszDudek replied:
fair point, I wanted to be conservative in order not to introduce regression but if you say that all video types are supported I'm ok with your suggestion.

dgozman reviewed (commented):
Thank you for the PR!

RemigiuszDudek committed:
Update packages/html-reporter/src/testResultView.tsx

fair point, I wanted to be conservative in order not to introduce regression but if you say that all video types are supported I'm ok with your suggestion.

Co-authored-by: Dmitry Gozman <dgozman@gmail.com>
Signed-off-by: Remigiusz Dudek <remigiusz_dudek@tlen.pl>

dgozman reviewed (approved):
Thank you, this looks great, merging in!
